app/ante/ante.go

```diff
@@ -94,28 +97,48 @@ func NewAnteHandler(options ethante.HandlerOptions) (sdk.AnteHandler, error) {
 		// handle as totally normal Cosmos SDK tx
 		switch tx.(type) {
 		case sdk.Tx:
-			found := false
-			for _, msg := range tx.GetMsgs() {
-				switch msg.(type) {
-				// treat these three msg types differently because they might call EVM which results in massive gas consumption
-				// For these two msg types, we don't check gas limit by using a different ante handler
-				case *cctxtypes.MsgGasPriceVoter, *cctxtypes.MsgVoteOnObservedInboundTx:
-					found = true
-					break
-				case *stakingtypes.MsgCreateValidator:
-					if ctx.BlockHeight() == 0 {
-						found = true
-						break
-					}
+			anteHandler = newCosmosAnteHandler(options)
+			if len(tx.GetMsgs()) != 1 {
+				break
+			}
```

Comment:
Need comment why this break when len  != 1

Reply:
Why do we need to limit all transactions to single message , can we add this check after we are able to determine that the TX contains a system message .

Reply:
It just break and use the regular anteHandler. So system message need to be unique in the transaction to use the no gas ante handler, which I think is ok.